apis/datacatalog/v1alpha1/tag_types.go

```diff
@@ -0,0 +1,126 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/apis/k8s/v1alpha1"
+)
+
+var DataCatalogTagGVK = GroupVersion.WithKind("DataCatalogTag")
+
+// Parent defines the parent resource for this DataCatalogTag
+type DataCatalogTagParent struct {
+	// Required. Reference to the DataCatalogEntry that owns this Tag.
+	// The entry must be in the same project and location as the tag.
+	// +required
+	EntryRef *EntryRef `json:"entryRef,omitempty"`
+}
+
+// DataCatalogTagSpec defines the desired state of DataCatalogTag
+// +kcc:proto=google.cloud.datacatalog.v1.Tag
+type DataCatalogTagSpec struct {
+	// Specifies the parent resource where this DataCatalogTag resides.
+	// +required
+	DataCatalogTagParent `json:",inline"`
+
+	// The DataCatalogTag name. If not given, the metadata.name will be used.
+	// +optional
+	ResourceID *string `json:"resourceID,omitempty"`
+
+	// Required. The resource name of the tag template this tag uses.
+	//
+	//  This field cannot be modified after creation.
+	// +kcc:proto:field=google.cloud.datacatalog.v1.Tag.template
+	// +required
+	// +immutable
+	// +kubebuilder:validation:Required
```

Comment:
Is this the same as `// +required` (which is already added to the field)

Reply:
doesnt seem to be. removing.